checks/kubernetes_resource.go

```diff
@@ -277,3 +299,54 @@ func (c *KubernetesResourceChecker) validate(ctx *context.Context, check v1.Kube
 
 	return nil
 }
+
+func deleteResource(ctx *context.Context, resource unstructured.Unstructured, waitForDelete bool) error {
+	ctx.Logger.V(4).Infof("deleting resource (%s/%s/%s)", resource.GetKind(), resource.GetNamespace(), resource.GetName())
+
+	rc, err := ctx.Kommons().GetRestClient(resource)
+	if err != nil {
+		return fmt.Errorf("failed to get rest client for (%s/%s/%s): %w", resource.GetKind(), resource.GetNamespace(), resource.GetName(), err)
+	}
+
+	namespace := utils.Coalesce(resource.GetNamespace(), ctx.Namespace)
+	deleteOpt := &metav1.DeleteOptions{
+		GracePeriodSeconds: lo.ToPtr(int64(0)),
+		PropagationPolicy:  lo.ToPtr(metav1.DeletePropagationOrphan),
+	}
+	if _, err := rc.DeleteWithOptions(namespace, resource.GetName(), deleteOpt); err != nil {
+		var statusErr *apiErrors.StatusError
+		if errors.As(err, &statusErr) {
+			switch statusErr.ErrStatus.Code {
+			case 404:
+				return nil
+			}
+		}
+
+		return fmt.Errorf("failed to delete resource %s: %w", resource.GetName(), err)
+	}
+
+	if !waitForDelete {
+		return nil
+	}
+
+	// Poll the resource until it's deleted
+	ticker := time.NewTicker(2 * time.Second)
```

Comment:
This should probably be split up, delete all the resources, then poll all the resources otherwise we wait for 1 resource to be deleted, before deleting the next,

Reply:
Which means we will poll all resources concurrently -  If we delete all resources first and then start polling:

1) If there are many resources, the first resources may have already been deleted by the time we start polling
2) After finding the first non-deleted resources we can sleep instead of polling them all, as the resources are deleted, remove them from the list of polling and only check the first item in the list-